usb/dev.c

```diff
@@ -0,0 +1,437 @@
+/*
+ * Phoenix-RTOS
+ *
+ * USB Host Device
+ *
+ * Copyright 2021 Phoenix Systems
+ * Author: Maciej Purski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+
+#include <errno.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/threads.h>
+#include <sys/list.h>
+
+#include <usb.h>
+
+#include "usbhost.h"
+#include "dev.h"
+#include "drv.h"
+#include "hcd.h"
+#include "hub.h"
+
+#define USBDEV_BUF_SIZE 0x200
+
+struct {
+	handle_t lock;
+	handle_t cond;
+	char *ctrlBuf;
+	char *setupBuf;
+} usbdev_common;
+
+
+void usb_devCtrlFinished(usb_transfer_t *t)
+{
+	condSignal(usbdev_common.cond);
```

Comment:
Hmm, but t.finished is not set? Where is it set and why not here?

Reply:
I know that these are volatile, but this variable is used as conditional variable - should be accessed only under lock. So setting t.finished etc. should be moved to this repository so you'll be able to use the mutex